### ISSUE TITLE
[mediaqueries-5] Consider specifying privacy guidelines for user agents which automatically toggle prefers-color-scheme based on user location data

### ISSUE THREAD

**Issue by Ajedi32** [w3c/csswg-drafts#4404]
Currently, the Media Queries Level 5 spec has this to say about the conditions a user agent may decide to take into account when determining what value to use for [prefers-color-scheme](https://drafts.csswg.org/mediaqueries-5/#prefers-color-scheme):

> The method by which the user expresses their preference can vary. It might be a system-wide setting exposed by the Operating System, or a setting controlled by the User Agent.

One common use case I've seen pop up lately is users wanting their system-wide dark mode setting to automatically toggle based on location-dependent data, such as the position of the sun relative to their current geolocation (i.e. they want to automatically turn on "dark mode" at dusk). If implemented naively without taking privacy into account, such a feature combined with prefers-color-scheme could potentially reveal the user's longitude to all websites with a remarkable degree of precision. (Multiple readings over the course of a year might also be able to determine latitude to some extent.)

In my opinion, this particular privacy concern seems both common enough and significant enough that it may be worth calling out explicitly in the spec to ensure it is properly addressed by user agents which choose to implement that feature.

This is also related to #3488

**Comment by FlorianWendelborn** [comment]
IMO this should be solved at the OS level. Great catch though

EDIT: To elaborate, why the OS would be the right place to solve this, end users expect their whole OS to switch between dark and light mode at the same time. It would be at least a minor annoyance, and possibly even mass-reported as a bug, if their browser and/or loaded website wouldn't switch at the same time. The only possible place where this can be solved without causing this UX issue is in the operating system itself. Additionally, the privacy benefits would then apply to all other apps besides supported browsers. Especially apps that do not have permissions to access the user's location.

In case any browser developer who wants to implement this privacy improvement reads this, please consider to additionally:
- check if the OS is known to already take care of this, and disable your own mitigations
- offer a toggle to disable the mitigation, so that users can decide to prefer the better UX instead of the slightly better privacy

**Comment by tabatkins** [comment]
OSes tend to expose much more fine-grained privacy-sensitive information about the user to programs already; the sort of privacy concerns we worry about with MQs usually aren't on the radar there. I think it's quite reasonable to bring this up as a concern in the query and suggest, for example, coarse time-bucketing.

**Comment by FlorianWendelborn** [comment]
@tabatkins yeah, I can definitely see that some operating systems may decide that this is a wontfix for them. I edited my initial comment to elaborate why I think the OS is the right place to solve this issue in.
